data/selinux/snappy.fc

```diff
@@ -18,30 +18,38 @@
 
 
 HOME_DIR/snap(/.*)?			gen_context(system_u:object_r:snappy_home_t,s0)
+/root/snap(/.*)?			gen_context(system_u:object_r:snappy_home_t,s0)
 
-
-/usr/bin/snap			--	gen_context(system_u:object_r:snappy_exec_t,s0)
-/usr/bin/snapctl		--	gen_context(system_u:object_r:snappy_exec_t,s0)
+/usr/bin/snap			--	gen_context(system_u:object_r:snappy_cli_exec_t,s0)
+/usr/bin/snapctl		--	gen_context(system_u:object_r:snappy_cli_exec_t,s0)
 
 ifdef(`distro_redhat',`
+/usr/libexec/snapd/snapctl		--	gen_context(system_u:object_r:snappy_cli_exec_t,s0)
+/usr/libexec/snapd/snap-confine		--	gen_context(system_u:object_r:snappy_confine_exec_t,s0)
+/usr/libexec/snapd/snap-update-ns		--	gen_context(system_u:object_r:snappy_mount_exec_t,s0)
+/usr/libexec/snapd/snap-discard-ns		--	gen_context(system_u:object_r:snappy_mount_exec_t,s0)
 /usr/libexec/snapd/.*		--	gen_context(system_u:object_r:snappy_exec_t,s0)
```

Comment:
Would the ordering matter here? Should the generic entry be first, and then the specific entries after?

Reply:
No, the order doesn't matter AFAIK. I'd leave the `/usr/libexec/snapd/.*` last to not confuse people.